web/app/view/dialog/DeviceController.js

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2017 Anton Tananaev (anton@traccar.org)
+ * Copyright 2017 Andrey Kunitsyn (andrey@traccar.org)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+Ext.define('Traccar.view.dialog.DeviceController', {
+    extend: 'Traccar.view.dialog.BaseEditController',
+    alias: 'controller.device',
+
+    init: function () {
+        if (Traccar.app.getUser().get('admin')) {
+            this.lookupReference('disabledField').setHidden(false);
```

Comment:
Do we have a check on the server side to make sure regular user can't enable or disable his devices?

Reply:
Regular user do not have access to disabled devices at all. `checkDevice` uses `getUserItems` function that filters disabled devices.
https://github.com/tananaev/traccar/blob/master/src/org/traccar/database/PermissionsManager.java#L302

Reply:
OK, I guess it good enough.